base/task/sequence_manager/task_queue_impl.cc

```diff
@@ -518,7 +513,58 @@ void TaskQueueImpl::PostImmediateTaskImpl(PostedTask task,
   // when it computes what continuation (if any) is needed.
   if (should_schedule_work)
     sequence_manager_->ScheduleWork();
+}
 
+void TaskQueueImpl::PostImmediateTaskImplUnordered(PostedTask task,
+                                          CurrentThread current_thread) {
+  // Use CHECK instead of DCHECK to crash earlier. See http://crbug.com/711167
+  // for details.
+  CHECK(task.callback);
+  CHECK(recordreplay::AreEventsDisallowed("unordered-tasks") ||
+        recordreplay::AreEventsPassedThrough("unordered-tasks"));
+
+  recordreplay::AutoPassThroughEvents passthrough;
```

Comment:
Why do we need to pass through events here given they are already disallowed / passed through?  This should at least have a comment, and it would be really nice if we could avoid it entirely.